src/payment/strategies/bluesnapv2/bluesnapv2-payment-strategy.spec.ts

```diff
@@ -0,0 +1,195 @@
+        orderActionCreator = new OrderActionCreator(
+            new OrderRequestSender(createRequestSender()),
+            new CheckoutValidator(new CheckoutRequestSender(createRequestSender())),
+            new SpamProtectionActionCreator(createSpamProtection(createScriptLoader()))
+        );
+        paymentActionCreator = new PaymentActionCreator(
+            new PaymentRequestSender(createPaymentClient()),
+            orderActionCreator,
+            new PaymentRequestTransformer()
+        );
+        blueSnapV2PaymentPageLoader = new BlueSnapV2PaymentPageLoader();
+        blueSnapV2PaymentPageLoader.initialize({
+            addFrame: jest.fn(),
+            removeFrame: jest.fn(),
+        });
+        finalizeOrderAction = of(createAction(OrderActionType.FinalizeOrderRequested));
+        initializeOffsitePaymentAction = of(createAction(PaymentActionType.InitializeOffsitePaymentRequested));
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+
+        options = { methodId: 'foobar' };
+        payload = merge(getOrderRequestBody(), {
+            payment: {
+                methodId: options.methodId,
+                paymentData: null,
+            },
+        });
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(orderActionCreator, 'finalizeOrder')
+            .mockReturnValue(finalizeOrderAction);
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'initializeOffsitePayment')
+            .mockReturnValue(initializeOffsitePaymentAction);
+
+        strategy = new BlueSnapV2PaymentStrategy(store, orderActionCreator, paymentActionCreator, blueSnapV2PaymentPageLoader);
+    });
+
+    it('submits order with payment data', async () => {
+        await strategy.execute(payload, options);
+
+        expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(payload, options);
+        expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+    });
+
+    it('initializes offsite payment flow', async () => {
+        await strategy.execute(payload, options);
+
+        expect(paymentActionCreator.initializeOffsitePayment)
+            .toHaveBeenCalledWith(
+                options.methodId,
+                options.gatewayId,
+                'bluesnapv2_hosted_payment_page'
+            );
+        expect(store.dispatch).toHaveBeenCalledWith(
+            initializeOffsitePaymentAction,
+            {
+                queueId: expect.stringMatching(
+                    /^BlueSnapV2PaymentStrategy:initializeOffsitePayment:[a-z0-9]{7}$/
+                ),
+            }
+        );
+    });
+
+    it('returns cancel error if the user cancels flow', () => {
+        jest.spyOn(blueSnapV2PaymentPageLoader, 'loadPaymentPage')
+            .mockImplementation(() => Promise.reject(new PaymentMethodCancelledError()));
+
+        return expect(strategy.execute(payload))
```

Comment:
do you need to return?

Reply:
It says the doc of jest [here](https://jestjs.io/docs/en/asynchronous#resolves--rejects) 🤷‍♂️

Reply:
I think that's only for resolves/rejects

Reply:
Sorry, the code is outdated, now I have resolves and rejects 😅